src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerExecutor.cs

```diff
@@ -823,23 +852,23 @@ private void ExecuteBlockActionsCore<TBlockStartAction, TBlockAction, TNodeActio
                             operationBlockScope, operationBlocks, declaredSymbol, semanticModel.Compilation, AnalyzerOptions,
                             GetControlFlowGraph, declaredNode.SyntaxTree, filterSpan, isGeneratedCode, cancellationToken);
 
-                        // Catch Exception from the start action.
                         ExecuteAndCatchIfThrows(
                             startAction.Analyzer,
-                            data =>
-                            {
-                                data.action(data.context);
-                                data.blockEndActions?.AddAll(data.scope.OperationBlockEndActions);
-                                data.operationActions?.AddRange(data.scope.OperationActions);
-                            },
-                            (action: operationBlockStartAction.Action, context: operationStartContext, scope: operationBlockScope, blockEndActions: operationBlockEndActions, operationActions: operationActions),
+                            static data => data.operationBlockStartAction.Action(data.operationStartContext),
+                            (operationBlockStartAction, operationStartContext),
                             new AnalysisContextInfo(Compilation, declaredSymbol),
```

Comment:
note: this context info will be lifted out of the loop in another pr (when this all gets strongly typed).